src/tbp/monty/frameworks/config_utils/config_args.py

```diff
@@ -1153,6 +1154,197 @@ class FiveLMMontySOTAConfig(FiveLMMontyConfig):
+                        "hsv": np.array([0.1, 0.2, 0.2]),
+                        "principal_curvatures_log": np.ones(2),
+                    }
+                },
+                feature_weights={
+                    "patch": {
+                        "hsv": np.array([1, 0.5, 0.5]),
+                    }
+                }
+            }
+            ```
+          When we are constructing the config for learning module **i**, that entry
+          for `"patch"` will be replaced with `"patch_{i}"`. This is currently
+          done for three possible items in `learning_module_args`:
+          `"graph_delta_thresholds"`, `"tolerances"`, and `"feature_weights"`.
+
+
+    Args:
+        n_lms (int): Number of learning modules.
+        monty_class (type): Monty class.
+        learning_module_class (type): Learning module class.
+        learning_module_args (dict, optional): Arguments for learning modules.
+        sensor_module_class (type): Sensor module class.
+        sensor_module_args (dict, optional): Arguments for sensor modules.
+        motor_system_class (type): Motor system class.
+        motor_system_args (Mapping, optional): Arguments for motor system.
+        monty_args (Mapping, MontyArgs, optional): Arguments for monty.
+        connectivity_func (Callable[[int], Mapping], optional): Function that returns a
+            dictionary of connectivity matrices given a number of learning modules.
+            In particular, it must return a dictionary with keys "sm_to_agent_dict",
+            "sm_to_lm_matrix", "lm_to_lm_matrix", and "lm_to_lm_vote_matrix". Defaults
+            to `make_multi_lm_flat_dense_connectivity`.
+
+    Returns:
+        `MontyConfig`: complete monty config for multi-LM experiment.
+    """
+    # Make learning module configs.
+    if learning_module_args is None:
+        learning_module_args = {}
+    learning_module_configs = {}
+    for i in range(n_lms):
+        lm_args_i = copy.deepcopy(learning_module_args)
+        # Rename specs keyed with "patch" to "patch_{i}".
+        for name in ["graph_delta_thresholds", "tolerances", "feature_weights"]:
+            if name in lm_args_i:
+                spec = lm_args_i[name]
+                if "patch" in spec:
+                    spec[f"patch_{i}"] = spec.pop("patch")
+        learning_module_configs[f"learning_module_{i}"] = {
+            "learning_module_class": learning_module_class,
+            "learning_module_args": lm_args_i,
+        }
+
+    # Make sensor module configs.
+    if sensor_module_args is None:
+        sensor_module_args = {}
+    sensor_module_configs = {}
+    for i in range(n_lms):
+        sm_args_i = copy.deepcopy(sensor_module_args)
+        sm_args_i["sensor_module_id"] = f"patch_{i}"
+        sensor_module_configs[f"sensor_module_{i}"] = {
+            "sensor_module_class": sensor_module_class,
+            "sensor_module_args": sm_args_i,
+        }
+    sensor_module_configs["view_finder"] = {
```

Comment:
Maybe it's worth being able to specify these, i.e. with a separate "view_finder" args? (which otherwise default to these?)

Reply:
Done